app.rb

```diff
@@ -0,0 +1,199 @@
+require_relative './classes/musicalbum'
+require_relative './iofile/read_data'
+require_relative './iofile/save_data'
+require 'json'
+
+class App
+  def initialize
+    @books = []
+    @lables = []
+    @games = ReadData.new.read_games
+    @authors = []
+    @genres = []
+    list_all_stored_books
+    list_all_stored_lables
+    @musicalbums = ReadData.new.read_music_album
+  end
+
+  def quit_app
+    SaveData.new.save_music_album(@musicalbums)
+    SaveData.new.save_games(@games)
+  end
+
+  def add_book
+    puts 'Publisher:'
+    publish = gets.chomp
+    puts 'Cover State:'
+    cover = gets.chomp
+    puts 'publish Date (yyyy-dd-mm):'
+    date = gets.chomp.to_i
+    book = Book.new(publish, cover, date)
+    @books.push(book)
+    puts 'Would you like to add lable? (1)- Yes // (2)- No'
+    options = gets.chomp.to_i
+    return unless options == 1
+
+    puts 'Choose a title for lable:'
+    title = gets.chomp
+    puts 'Choose a color for the lable:'
+    color = gets.chomp
+    lable = Lable.new(title, color)
+    @lables.push(lable)
+    all_lables_books
+    puts 'Your Book Added Successfully✅'
+  end
+
+  def all_lables_books
+    bkjson = []
+    @books.each do |bk|
+      bkjson.push({ publisher: bk.publisher, cover_state: bk.cover_state, publish_date: bk.publish_date })
+    end
+    bookjson = JSON.generate(bkjson)
+    File.write('book.json', bookjson)
+
+    labjson = []
+    @lables.each do |lab|
+      labjson.push({ title: lab.title, color: lab.color })
+    end
+    lablejson = JSON.generate(labjson)
+    File.write('lable.json', lablejson)
+  end
+
+  def list_all_books
+    if @books.empty?
+      puts "Sorry, We do not have any book\n\n"
+    else
+      @books.each do |bk|
+        puts "Publisher:#{bk.publisher} Cover State:#{bk.cover_state} Publish Date:#{bk.publish_date}\n"
+      end
+    end
+  end
+
+  def list_all_lables
+    if @lables.empty?
+      puts "Sorry, We do not have any lables\n\n"
+    else
+      @lables.each { |lab| puts "Lable Title:#{lab.title} color:#{lab.color}\n" }
+    end
+  end
+
+  def list_all_stored_books
+    if File.exist?('book.json') && !File.zero?('book.json')
+      bookfile = File.open('book.json')
+      bookjson = bookfile.read
+      JSON.parse(bookjson).map do |bk|
+        example = Book.new(bk['publisher'], bk['cover_state'], bk['publish_date'])
+        @books.push(example)
+      end
+      bookfile.close
+    else
+      File.new('book.json', 'w')
+    end
+  end
+
+  def list_all_stored_lables
+    if File.exist?('lable.json') && !File.zero?('lable.json')
+      lablefile = File.open('lable.json')
+      lablejson = lablefile.read
+      JSON.parse(lablejson).map do |lab|
+        lablexample = Lable.new(lab['title'], lab['color'])
+        @lables.push(lablexample)
+      end
+      lablefile.close
+    else
+      File.new('lable.json', 'w')
+    end
+  end
+
+  def add_game
+    puts 'publish Date (yyyy-dd-mm):'
+    publish_date = gets.chomp.to_i

```

Comment:
- [ ] @team as previously requested kindly ensure you add a method to verify the date inputs format, when the user add new dates